server/game/cards/02_SHD/units/FrontierTrader.ts

```diff
@@ -0,0 +1,30 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { ZoneName } from '../../../core/Constants';
+
+export default class FrontierTrader extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '8095362491',
+            internalName: 'frontier-trader'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addWhenPlayedAbility({
+            title: 'You may return a resource you control to your hand. If you do, you may put the top card of your deck into play as a resource.',
+            optional: true,
+            targetResolver: {
+                zoneFilter: ZoneName.Resource,
+                immediateEffect: AbilityHelper.immediateEffects.returnToHand()
```

Comment:
Need to add `controller: RelativePlayer.Self` or else it might be possible to select the opponent's resources as well